Updated to install and configure shibboleth SP2

blarghmatey opened the pull request:
Installs the shibboleth service provider and configures Nginx to be able to interface with it via FastCGI. uWSGI is used
to serve the shibboleth CGI scripts via the fastcgi protocol. The shibboleth configuration is rendered from pillar data.

blarghmatey committed:
Updated to install and configure shibboleth SP2

Installs the shibboleth service provider and configures Nginx to be able to interface with it via FastCGI. uWSGI is used
to serve the shibboleth CGI scripts via the fastcgi protocol. The shibboleth configuration is rendered from pillar data.

blarghmatey committed:
Added option to generate sp-cert

shaidar commented on nginx-shibboleth/install.sls:
```diff
@@ -1,10 +1,53 @@
+          - git: clone_nginx_headers_module
+          - git: clone_nginx_shibboleth_plugin
+      - require_in:
+          - file: copy_shibboleth_includes_to_nginx_path
```
Where is this file coming from?

blarghmatey replied:
The string is the id of the `file.copy` state, but the file being copied is part of the git repository for the nginx_http_shibboleth plugin.

shaidar commented on nginx-shibboleth/config.sls:
```diff
@@ -1,15 +1,33 @@
+    - name: shib-keygen
+    - creates: /etc/shibboleth/sp-cert.pem
+{% else %}
+{% for secret in ['key', 'cert'] %}
```
How is that finding ['key', 'cert'] since in the [pillar.example](https://github.com/mitodl/nginx-shibboleth-formula/blob/blarghmatey/shibboleth_install/pillar.example#L5) those are under `secrets`?

blarghmatey replied:
Those are just string variables used for interpolation in  the file.managed states.

blarghmatey committed:
Updated to use supervisor instead of uWSGI

Updated the formula to use supervisor for managing the FCGI processes for interfacing the Shibboleth SP with Nginx since
uWSGI doesn't appear to be able to properly handle the FCGI protocol.

Also updated the example pillar to be more comprehensive.

blarghmatey committed:
Updated tests and service management

blarghmatey commented:
@shaidar this should be ready for review now.

@singingwolfboy if you have a few minutes to spare it might be helpful for you to take a brief look to see if there are any pieces that I missed since I was largely working from your (very helpful) notes :smile:

shaidar commented on nginx-shibboleth/tests/test_config.sls:
```diff
@@ -0,0 +1,15 @@
+        expected: _shibd
+        comparison: eq
+    - mode:
+        expected: 420
```
Is this right as I don't believe it matches the permissions set [here](https://github.com/mitodl/nginx-shibboleth-formula/blob/blarghmatey/shibboleth_install/nginx-shibboleth/config.sls#L18)? Additionally, does [this](https://github.com/mitodl/nginx-shibboleth-formula/blob/blarghmatey/shibboleth_install/nginx-shibboleth/config.sls#L18) need to be 0644?

blarghmatey committed:
Added support for RHEL and parameterized supervisord user

shaidar commented on README.rst:
```diff
@@ -24,7 +24,10 @@ TODO - add description of this state
 ``nginx-shibboleth.conf``
```
Shouldn't this be `nginx-shibboleth.config`?

shaidar commented on nginx-shibboleth/install.sls:
```diff
@@ -1,10 +1,68 @@
+
+install_service_provider:
   pkg.installed:
     - pkgs: {{ nginx_shibboleth.pkgs }}
```
`nginx_shibboleth.pkgs` should be `nginx_shibboleth.pkg` or the key in `map.jinja` needs to be changed

shaidar commented on nginx-shibboleth/config.sls:
```diff
@@ -1,15 +1,49 @@
+    - name: /etc/shibboleth/sp-{{ secret }}.pem
+    - mode: 0600
+    - user: _shibd
+    - group: _shibd
```
How are `user` and `group` going to work for RedHat since it appears that you're using `shibd` and `root`

shaidar commented on nginx-shibboleth/config.sls:
```diff
@@ -1,15 +1,49 @@
+    - mode: 0600
+    - user: _shibd
+    - group: _shibd
+    - contents_pillar: nginx-shibboleth:secrets:{{ secret }}
```
Not exactly sure how this will be rendered as I don't see `secrets` defined in `map.jinja` or the pillar overrides

blarghmatey replied:
If the values are present in the pillar at nginx-shibboleth:secrets:key/cert then the contents of those values will be rendered out. Otherwise it will generate a set of certificate and key.

shaidar commented on nginx-shibboleth/files/shibboleth_sp_fcgi.conf:
```diff
@@ -0,0 +1,17 @@
+socket=unix:///run/shibauthorizer.sock
+socket_owner=_shibd:www-data
+socket_mode=0660
+user=_shibd
```
RedHat `shibd` user?

blarghmatey committed:
Updated versions and fixed compatibility issues with RHEL distros

- Added templating of users for shibboleth supervisor processes and files
- Simplified logic for handling sp-cert and sp-key
- Updated version to reflect actual release dates

shaidar commented on README.rst:
```diff
@@ -16,15 +16,19 @@ Available states
+provider via the FastCGI protocol.
 
-``nginx-shibboleth.conf``
+``nginx-shibboleth.config_``
```
Does the underscore in `config` serve some secret/special purpose?

shaidar commented:
👍
